snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2TransactionContext.java

```diff
@@ -177,11 +180,21 @@ void add(Collection<SnomedComponent> componentChanges, Multimap<Class<? extends
 					// new component, add to new components and register row for import
 					newComponents.put(rf2Component.getId(), createIdOnlyDoc(rf2Component.getId(), type));
 					componentsToImport.add(rf2Component);
+					if (rf2Component instanceof SnomedCoreComponent) {
+						if (rf2Component.getEffectiveTime() == null) {
+							idsToRegister.add(rf2Component.getId());
+						} else {
+							idsToPublish.add(rf2Component.getId());
+						}
+					}
 				} else if (existingObject instanceof SnomedDocument && rf2Component instanceof SnomedComponent) {
 					final SnomedComponent rf2Row = (SnomedComponent) rf2Component;
 					final SnomedDocument existingRow = (SnomedDocument) existingObject;
 					if (rf2Row.getEffectiveTime() == null || EffectiveTimes.getEffectiveTime(rf2Row.getEffectiveTime()) > existingRow.getEffectiveTime()) {
 						componentsToImport.add(rf2Component);
+						if (existingRow instanceof SnomedComponentDocument && rf2Row.getEffectiveTime() !=null && !existingRow.isReleased()) {
```

Comment:
While this looks okay from the CIS point of view, my question is, does the system also apply the `released` `false` -> `true` state transition? (I vaguely remember yes, but need a confirmation)

Reply:
Done here: https://github.com/b2ihealthcare/snow-owl/blob/6b7e0eaf0e8ac4dfa84b1826ba3ab4e766f6c2c9/snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/store/SnomedComponentBuilder.java#L108-L114

`SnomedComponentBuilder#init` is called here: https://github.com/b2ihealthcare/snow-owl/blob/6b7e0eaf0e8ac4dfa84b1826ba3ab4e766f6c2c9/snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2TransactionContext.java#L256-L257

Reply:
Correct `released` flag transitions are ensured by PR https://github.com/b2ihealthcare/snow-owl/pull/802.

Reply:
Thank you @apeteri for checking. 🙇

Reply:
Also verified by a test case in b0280b4 below.